Space

HaoK opened the pull request.

davidfowl commented on src/overrides/Helix.Sdk/tools/aspnet/AspNetHelix.targets:
```diff
@@ -0,0 +1,195 @@
+    </ItemGroup>
+  </Target>
+
+  <Target Name="CreateAspNetTestWorkItems"
```
What would it take to use the xunit work item logic?

HaoK replied:
I'm still mucking with msbuild to get things working, nate helped me figure out the appropriate build goop back in the day, and of course how we do things is significantly different than how things are setup in this repo. I'm probably going to Frankenstein our test runner into the existing CreateXunitWorkItem task as a first step before trying to figure out how to get our stuff working over here.

HaoK replied:
Also we have the logic to filter out specific tests based on the OS queue directly in our gather step.  Basically we gather all tests for each helix queue, while CreateXunitWorkItem is closer to just the innermost _CreateHelixWorkItem target for us.

davidfowl replied:
I see. What we have is still much more advanced than what’s in helix. This filtering should be hoisted into helix maybe

HaoK replied:
The other thing we rely heavily on is injecting an environment variable with the helix queue name, since our tests skip conditionally based on the queue name, maybe this is something that could be pushed into the helix layer, like a HELIX_QUEUE environment variable.  This doesn't look like something I can easily do inside of CreateXunitWorkItem

HaoK commented:
Okay finally some progress, windows queues are working with our test runner, now 

  https://helix.dot.net/api/2019-06-17/jobs/804216ce-bd6a-454c-befb-a5a8b7dea274/workitems/AspNetTestProject1.dll/files/console.e95657ab.log

I'm not sure what's different about how our files are getting sent to the non windows machines, but our scripts seem to be failing with a line ending issue almost on osx/ubunutu:

```
Executed on dci-macpro-20.local
+ export PATH=/tmp/helix/working/ABC20959/p/dotnet:/usr/bin:/bin:/usr/sbin:/sbin
+ PATH=/tmp/helix/working/ABC20959/p/dotnet:/usr/bin:/bin:/usr/sbin:/sbin
+ ./runtests.sh AspNetTestProject1.dll 5.0.100-preview.2.20120.3 3.1.0 _HelixFriendlyNameTargetQueue x64 false DotnetEfPackageVersion
env: bash\r: No such file or directory
+ export _commandExitCode=127

```

HaoK commented:
Okay I hacked our test runner into the existing xunit task, I deleted all the extra stuff for now, basically just kept the test runner and the bootstrapping scripts.  Most of the complexity we have is needed for how we build things.  Since things here are setup to just use the publish directory without building, I just deleted all that stuff for now.  

So the main additional things we do are:
1. install specific prerelease versions of sdk and runtime, not just a single sdk/runtime version
2. setup a bunch of environment variables for our tests (these are mostly specific to aspnet tests, except for the helix queue name which seems like a good default)
3. some test results and log moving around to the right place for uploading

davidfowl commented on eng/helix/content/default.runner.json:
```diff
@@ -0,0 +1,4 @@
+{
```
Oh I forgot about this!

davidfowl commented on src/overrides/Helix.Sdk/CreateXUnitWorkItems.cs:
```diff
@@ -46,6 +46,11 @@ public class CreateXUnitWorkItems : BaseTask
+        /// <summary>
+        /// Run in aspnet mode
+        /// </summary>
+        public bool AspNetMode { get; set; }
```
Is ASP.NET Core downloading our custom build shared framework and running tests on it? What's the summary of what this does compared to the built in?

HaoK replied:
I just used this flag as a switch to hijack the CreateXunitTask, basically in my branch we are running everything through the aspnet test runner now, so its running against preview sdk and on our scripts, its not using the default xunit-runner anymore

HaoK replied:
I just swapped out the command with our test runner stuff with hardcoded 5.0-preview2 sdk: see https://github.com/dotnet/helix-diagnostic-prototype/blob/haok/src/overrides/Helix.Sdk/CreateXUnitWorkItems.cs#L150

HaoK replied:
I can't use the 5.0 app runtime because the tests projects are targetting 3.1
